### PR TITLE
Added RFC-0190 Mempool 

### PR DESCRIPTION
**Pull request by neonknight64**
## Description
- Added RFC-0190 Mempool
- Modified Mempool glossary item
- Added new glossary items for Transaction Pool, Pending Pool, Orphan pool and Reorg Pool
- Fixed Glossary issues: items sorted according to alphabetic order and fixed some formatting issues

## Motivation and Context
Mempool was a missing RFC for the Base Layer
 
## How Has This Been Tested?
No tests required

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [x] Documentation

## Checklist:

* [x] I'm merging against the `development` branch
* [x] I ran `cargo-fmt --all` before pushing
* [x] My change requires a change to the documentation.
* [x] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: RFC/src/RFC-0190_Mempool.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+reorganization occurs and these transactions have to be restored to the Transaction Pool so that they can be included 
+in future blocks.
+
+### Prioritizing Unconfirmed Transactions
+
+The maximum storage capacity used for storing unconfirmed transactions by the Mempool and each of its component pools 
+can be configured. If a new transaction is received and the storage capacity limits have been reached, then 
+transactions are prioritized according to the transaction priority metric. The transaction priority metric consist 
+of the fees and the maturity of the UTXOs being spent by the transaction and should be applied to determine the priority 
+of each transaction in the Mempool. As the allocated storage space of the Mempool become limited only transactions
```
Comment (sdbondi) [inline RFC/src/RFC-0190_Mempool.md]:
```suggestion
of each transaction in the Mempool. As the allocated storage space of the Mempool becomes limited, the transactions with the highest priority are kept in the pool. 
```

^ Do not commit this directly as it applies to the following line too

File: RFC/src/RFC-0190_Mempool.md [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+transactions are prioritized according to the transaction priority metric. The transaction priority metric consist 
+of the fees and the maturity of the UTXOs being spent by the transaction and should be applied to determine the priority 
+of each transaction in the Mempool. As the allocated storage space of the Mempool become limited only transactions 
+with the highest priority are kept in the pool. The lowest priority transactions are discarded to make room for 
+higher priority incoming transactions.
+
+The transaction priority metric has the following behavior:
+ - Transactions spending UTXOs with higher block height maturity MUST be prioritized over transactions spending UTXOs 
+ with lower block height maturity.
+ - Transactions with higher fees per transaction message size MUST be prioritized over lower fee transactions.
```
Comment (sdbondi) [inline RFC/src/RFC-0190_Mempool.md]: Should this be a `SHOULD`? It won't break anything if nodes take the lower fees, though obviously it's in their best interest to do so.

File: RFC/src/RFC-0190_Mempool.md [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+On the initial startup of the Mempool, the complete state of the Mempool that consists of the Transaction Pool, Pending 
+Pool, Orphan Pool and Reorg Pool can be requested and downloaded from the connected peers. A memory pool typically 
+doesn't make use of persistent storage but could be configured to keep a backup of the last known state. If an existing 
+Mempool state is locally available then a more efficient update process can be performed by requesting only the 
+unconfirmed transactions missing from the current Mempool state. When no state is available then the entire Mempool 
+state must be downloaded from the connected peers. During downloading or updating of the Mempool state, the validity 
+of all transaction in the pool must be verified and the priority of each transaction must be calculated. 
+
+Functional behavior required for sharing and updating of the Mempool state, and propagation of transactions between peers:
+- All Verified transaction MUST be propagated to neighboring peers.
```
Comment (sdbondi) [inline RFC/src/RFC-0190_Mempool.md]:
```suggestion
- All verified transactions MUST be propagated to neighboring peers.
```

File: RFC/src/RFC-0190_Mempool.md [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+- Duplicate transactions MUST NOT be propagated to peers.
+- Unvalidated or invalid transactions MUST NOT be propagated to peers.
+- Verified transactions that were discarded due to low priority levels MUST be propagated to peers.
+- The Mempool MUST have an interface that can be used by neighboring peers to query the content and state of the memory 
+pool.
+- It MUST have a mechanism that enables peers to download the memory pool state in full or in part.
+- It MUST accept all transactions received from peers but MAY decide to discard low priority transactions.
+- It MUST allow Wallets to track incoming payments by monitoring new transaction that has been added to the Mempool.
```
Comment (sdbondi) [inline RFC/src/RFC-0190_Mempool.md]:
```suggestion
- It MUST allow wallets to track payments by monitoring that a particular transaction has been added to the Mempool.
```

File: RFC/src/RFC-0190_Mempool.md [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+their time-locks or the time-locks of the input UTXOs to expire before they can be processed and included into new 
+blocks. All transactions in the Pending Pool have been verified and passed all checks except their own time-lock has 
+not yet expired or some of the UTXOs that will be spent have time-lock restrictions that are not yet valid. Once the 
+transactions time-lock or the time-locks on the UTXOs have expired then the Pending Pool transactions can be moved 
+to the Transaction Pool for inclusion in future blocks.  
+
+Functional behavior required of the Pending Pool:
+- Once the transaction time-lock or UTXO time-lock restricting the processing of a transaction has expired then the 
+pending transaction MUST be moved to the Transaction Pool.
+- Pending transactions that have matured past the discard threshold MUST be discarded and removed from the Pending Pool.
```
Comment (sdbondi) [inline RFC/src/RFC-0190_Mempool.md]: Only applicable for the reorg pool and maybe the orphan pool?

File: RFC/src/Glossary.md [hunk 6]
```diff
@@ -52,67 +60,84 @@ A sequence of tari [block]s. Each block contains a hash of the previous valid bl
...
-[currenthead]: #currenthead	"The last valid block of the longest chain"
-
-The last [block] of the base layer that represents the latest valid block. This [block] must be from the longest proof of work chain to be the current head.
 
-## Checkpoint
+## Blockchain state
+[blockchainstate]: #blockchainstate	"This is a snapshot of how the blockchain looks"
 
-[checkpoint]: #checkpoint "A summary of the state of a Digital Asset that is recorded on the base layer"
+This is a complete snapshot of the blockchain at a spesific block height. This means pruned [utxo], complete set of kernals and headers up to that block height from genesis block is known.
```
[the remark this answers is not in the document]
Comment (neonknight64) [reply RFC/src/Glossary.md]: Change applied

File: RFC/src/Glossary.md [hunk 7]
```diff
@@ -52,67 +60,84 @@ A sequence of tari [block]s. Each block contains a hash of the previous valid bl
...
 ## Creator Assigned Mode
+[creator assigned mode]: #creator-assigned-mode
 
 An asset runs in creator-assigned mode when _every_ validator node in a validator committee is a [Trusted Node].
 
 
+## Current head
+[currenthead]: #currenthead	"The last valid block of the longest chain"
+
+The last [block] of the base layer that represents the latest valid block. This [block] must be from the longest proof of work chain to be the current head.
```
[the remark this answers is not in the document]
Comment (neonknight64) [reply RFC/src/Glossary.md]: Change applied

File: RFC/src/Glossary.md [hunk 8]
```diff
@@ -189,82 +217,117 @@ A Mining Server is responsible for constructing new blocks by bundling transacti
...
 
-## SynchronisationStrategy
 
-The generalised approach for a [Base Node] to obtain the current state of the blockchain from the peer-to-peer network.
-Specific implementations may differ based on different trade-offs and constraints with respect to bandwidth, local
-network conditions etc.
+## Reorg Pool
+[reorg pool]: #reorg-pool "A backup pool in the Mempool for unconfirmed transactions that have been included in blocks"
+
+The reorg pool is part of the [mempool] and stores all [transaction]s that have been included in blocks in case a blockchain reorganization occurs and the transactions need to be restored to the [transaction pool].
```
[the remark this answers is not in the document]
Comment (neonknight64) [reply RFC/src/Glossary.md]: Change applied

File: RFC/src/RFC-0190_Mempool.md (hunk 1 above)
Comment (neonknight64) [reply RFC/src/RFC-0190_Mempool.md]: Change applied

File: RFC/src/RFC-0190_Mempool.md (hunk 2 above)
Comment (neonknight64) [reply RFC/src/RFC-0190_Mempool.md]: Change applied

File: RFC/src/RFC-0190_Mempool.md (hunk 3 above)
Comment (neonknight64) [reply RFC/src/RFC-0190_Mempool.md]: Change applied

File: RFC/src/RFC-0190_Mempool.md (hunk 4 above)
Comment (neonknight64) [reply RFC/src/RFC-0190_Mempool.md]: Change applied

File: RFC/src/RFC-0190_Mempool.md (hunk 5 above)
Comment (neonknight64) [reply RFC/src/RFC-0190_Mempool.md]: Correct, item was removed

**Pull request merged**